### PR TITLE
Allow inverting switch_as_x cover, lock, valve

### PR DESCRIPTION
**Pull request by emontnemery**
## Proposed change

Allow inverting switch_as_x cover, lock, valve

Core PR: https://github.com/home-assistant/core/pull/107535

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Review by bramkragten — DISMISSED** [review]

File: src/panels/config/entities/entity-registry-settings-editor.ts [hunk 1]
```diff
@@ -1194,20 +1198,18 @@ export class EntityRegistrySettingsEditor extends LitElement {
...
       const origEntityId = this.entry.options?.switch_as_x?.entity_id;
-      // remove current helper
-      await deleteConfigEntry(this.hass, this.helperConfigEntry.entry_id);
-
       if (!origEntityId) {
         // should not happen, guard for types
       } else {
-        const configFlow = await createConfigFlow(this.hass, "switch_as_x");
-        const configFlowResult = (await handleConfigFlowStep(
+        const configFlow = await createOptionsFlow(
```
Comment (bramkragten) [inline src/panels/config/entities/entity-registry-settings-editor.ts:1204]: Can we do the same for changing the switch as type? (Can be a different PR)

File: src/panels/config/entities/entity-registry-settings-editor.ts [hunk 2]
```diff
@@ -1194,20 +1198,18 @@ export class EntityRegistrySettingsEditor extends LitElement {
...
+        const configFlowResult = (await handleOptionsFlowStep(
           this.hass,
           configFlow.flow_id,
           {
-            entity_id: origEntityId,
             invert: this._switchAsInvert,
-            target_domain: this._switchAsDomain,
           }
         )) as DataEntryFlowStepCreateEntry;
         if (configFlowResult.result?.entry_id) {
```
Comment (bramkragten) [inline src/panels/config/entities/entity-registry-settings-editor.ts:1215]: We don't have to do this in this case? As the entity id stays the same there is no need to open a different more info dialog?

File: src/panels/config/entities/entity-registry-settings-editor.ts (hunk 1 above)
Comment (emontnemery) [reply src/panels/config/entities/entity-registry-settings-editor.ts:1204]: I would prefer not, because when changing the switch as type we want to delete the entity registry item etc.

File: src/translations/en.json [hunk 3]
```diff
@@ -1149,6 +1149,10 @@
           "wind_speed_unit": "Wind speed unit",
           "device_class": "Show as",
           "switch_as_x": "Show switch as",
+          "invert": "Invert state",
+          "invert_switch_as_cover_description": "Show as open when the switch is off",
+          "invert_switch_as_lock_description": "Show as locked when the switch is off",
+          "invert_switch_as_valve_description": "[%key:ui::dialogs::entity_registry::editor::invert_switch_as_cover_description%]",
```
Comment (bramkragten) [inline src/translations/en.json:1155]:
```suggestion
"invert" : {
  "label": "Invert state",
  "descriptions": {
  	"cover": "Show as open when the switch is off",
  	"lock": "Show as locked when the switch is off",
  	"valve": "Show as open when the switch is off",
  }
}
```

**Review by bramkragten — DISMISSED** [review]

**Review by bramkragten — APPROVED** [review]
